azkaban-common/src/main/java/azkaban/executor/container/watch/KubernetesWatch.java

```diff
@@ -75,9 +85,23 @@ public KubernetesWatch(ApiClient apiClient,
     this.podWatchEventListener = podWatchEventListener;
     this.podWatchParams = podWatchParams;
     this.apiClient = apiClient;
-    // no timeout for request completion
+    this.callTimeout =
+        azkProps.getInt(ContainerizedDispatchManagerProperties.KUBERNETES_WATCH_CALL_TIMEOUT,
+        DEFAULT_KUBERNETES_WATCH_CALL_TIMEOUT);
+    this.connectTimeout =
+        azkProps.getInt(ContainerizedDispatchManagerProperties.KUBERNETES_WATCH_CONNECT_TIMEOUT,
+            DEFAULT_KUBERNETES_WATCH_CONNECT_TIMEOUT);
+    this.readTimeout =
+        azkProps.getInt(ContainerizedDispatchManagerProperties.KUBERNETES_WATCH_READ_TIMEOUT,
+            DEFAULT_KUBERNETES_WATCH_READ_TIMEOUT);
+    // Setting read and call timeouts. So pod watch will be reinitialized.
     OkHttpClient httpClient =
-        this.apiClient.getHttpClient().newBuilder().readTimeout(0, TimeUnit.SECONDS).build();
+        this.apiClient.getHttpClient().newBuilder()
+            .protocols(Arrays.asList(Protocol.HTTP_2,Protocol.HTTP_1_1))
+            .connectTimeout(this.connectTimeout, TimeUnit.SECONDS)
```

Comment:
Is there a reason whey the client version upgrade needs to be clubbed with the changes for adding these timeouts ?
Ideally the timeouts can be added first (as they seem more urgent) and followed up with the version upgrade, the version upgrade changes will take a little more time to review as there are quite a few of those.

Reply:
Note:
To ensure our client's watch connection is not stale and is properly timed out/ refreshed, this upgrade isn't necessary. However, while we are upgrading the client to ensure we don't break compatibility in future K8s version upgrades, this timeout is introduced at the same time. We could possibly separate this change.